dataloaders/data_hybrid.py

```diff
@@ -0,0 +1,178 @@
+
+        for path in self.manager.args["output_paths"]:
+            file = uproot.open(path)
+            self.Y_chunk = file[self.manager.args["tree"]].arrays(self.manager.args["weights"], library="pd", entry_start = self.whichChunk*self.nec, entry_stop =(self.whichChunk+1)*self.nec).values
+            file.close()
+        ## Stacking of multiple paths not implemented yet
+
+        if(self.manager.args["shuffle"]):
+            self.X_chunk, self.Y_chunk = shuffle(self.X_chunk, self.Y_chunk)
+
+
+    def __getitem__(self, idx):
+        self.loadChunk()
+        if(idx == self.bic-1):
+            return self.X_chunk[idx*self.bs:], self.Y_chunk[idx*self.bs:]
+        else:
+            return self.X_chunk[idx*self.bs:(idx+1)*self.bs], self.Y_chunk[idx*self.bs:(idx+1)*self.bs]
+
+def DataloaderMultithread(data_manager, num_workers = 2):
+    file = uproot.open(data_manager.args["output_paths"][0])
+    len = file[data_manager.args["tree"]].num_entries # Total number of samples
+    file.close()
+
+    n_chunks = len // data_manager.args["chunk_entries"]
+    for chunk in range(n_chunks):
+        dataset = CustomDSHybridMultithread(data_manager, whichChunk=chunk)
+        generator = DataLoader(dataset, batch_size=1, shuffle=False, num_workers = num_workers)
+
+
+# path = "/data/deepmem_debug/tt_20evt.root"
+# path_weights = "/data/deepmem_debug/tt_20evt_weights.root"
+
+# file = uproot.open(path_weights)
+# a1 = file["t"].arrays(["bb_p4/fCoordinates/fCoordinates.fPt", "bb_p4/fCoordinates/fCoordinates.fEta", "bb_p4/fCoordinates/fCoordinates.fPhi", "bb_p4/fCoordinates/fCoordinates.fM"], library="pd").values.T
+# print(a1)
+# print(file["t"].show())
+
+# file.close()
+
+if(__name__ == "__main__"):
+    # This script is for testing the runtime of the dataloader.
+    # It has a similar structure to a common PyTorch training loop.
+
+    start = time.time()
+    data_manager = DataManager()
+
+    dataset = CustomDSHybrid(data_manager)
+    generator = DataLoader(dataset, batch_size=1, shuffle=False, num_workers = 0)
+
+    for batch_idx, data in enumerate(generator):
+        x, y = data
+        print(x.shape, y.shape, " -> batch ", batch_idx)
+
+    '''The following code tests the multithreaded version but is not completely object oriented'''
+    # start = time.time()
+    # data_manager = DataManager()
+
+    # file = uproot.open(data_manager.args["output_paths"][0])
+    # len = file[data_manager.args["tree"]].num_entries # Total number of samples
+    # file.close()
+    # n_chunks = len // data_manager.args["chunk_entries"]
+    # for epoch in range(1):
+    #     for chunk in range(n_chunks):
+    #         dataset = CustomDSHybridMultithread(data_manager, whichChunk=chunk)
+    #         generator = DataLoader(dataset, batch_size=1, shuffle=False, num_workers = 3)
+
+    #         for batch_idx, data in enumerate(generator):
+    #             x, y = data
+    #             print(x.shape, y.shape, " -> batch ", batch_idx)
```

Comment:
```suggestion
```

If there are things that you want to keep around to return to, I'd just keep them on another branch but I'd suggest not adding commented out code into `main`.